src/common/cockpitcloserange.c

```diff
@@ -0,0 +1,92 @@
+#include "cockpitcloserange.h"
+
+#include <assert.h>
+#include <dirent.h>
+#include <err.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/resource.h>
+
+static int
+closefd (int from, int max_fd, int fd)
+{
+  if (fd >= from && fd <= max_fd)
+    {
+      while (close (fd) < 0)
+        {
+          if (errno == EAGAIN || errno == EINTR)
+            continue;
+          if (errno == EBADF || errno == EINVAL)
+            break;
+          warnx ("couldn't close fd: %m");
+          return -1;
+        }
+    }
+
+  return 0;
+}
+
+int
+cockpit_close_range (int from, int max_fd, int flags)
+{
+  int open_max;
+  int fd;
+  int res = 0;
+
+  struct rlimit rl;
+
+  /* we want to keep the API compatible to glibc's upcoming close_range(), but don't implement flags */
+  assert (flags == 0);
```

Comment:
May as well assert that the upper range is -1 as well.  It lets us simplify logic some.

Reply:
Done.